sqlmesh/core/environment.py

```diff
@@ -228,6 +229,11 @@ class EnvironmentStatements(PydanticModel):
     jinja_macros: t.Optional[JinjaMacroRegistry] = None
 
 
+class EnvironmentCleanupTask(PydanticModel):
+    expired_environments: t.List[Environment]
+    filter_expr: exp.LTE
```

Comment:
This is IMHO not a good interface and a significant abstraction leak. 

Separately, why environment names are not enough? Why do we need full environment objects?